### PR TITLE
[12.0][MIG] account_invoice_check_total

### PR DESCRIPTION
**Pull request by rven**

### CODE REVIEW

File: account_invoice_check_total/models/account_invoice.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+                            inv.amount_total,
+                            precision_rounding=inv.currency_id.rounding
+                        ) != 0:
+                    raise UserError(_(
+                        'Please verify the price of the invoice!\n\
+                        The encoded total does not match the computed total.'))
+        return super(AccountInvoice, self).action_move_create()
+
+    @api.model
+    # pylint: disable=too-many-arguments
```
Comment (tafaRU) [inline account_invoice_check_total/models/account_invoice.py]: Is still needed this?

**Review by tafaRU — APPROVED** [review]
Code review and test locally :+1: 
Only one question.

**Review by ThomasBinsfeld — APPROVED** [review]

**Review by GillesTephaneMeyomesse — APPROVED** [review]

**Review by espo-tony — APPROVED** [review]

**Comment by ThomasBinsfeld** [comment]
@pedrobaeza This one is ready to be merged

**Review by mpanarin — APPROVED** [review]

**Comment by tafaRU** [comment]
Can we already merge altough the Odoo branch `12.0` is not yet released?

File: account_invoice_check_total/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    'name': 'Account Invoice Check Total',
+    'summary': """
+        Check if the verification total is equal to the bill's total""",
+    'version': '12.0.1.0.0',
+    'license': 'AGPL-3',
+    'author': 'Acsone SA/NV,Odoo Community Association (OCA)',
+    'website': 'https://acsone.eu/',
```
Comment (nikul-serpentcs) [inline account_invoice_check_total/__manifest__.py]: @rven Replace URL as per [this](https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py#L10)

File: account_invoice_check_total/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    'depends': [
+        'account',
+    ],
+    'data': [
```
Comment (nikul-serpentcs) [inline account_invoice_check_total/__manifest__.py:15]: Sequence as per [this](https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py#L27)

File: account_invoice_check_total/models/account_invoice.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        return super(AccountInvoice, self).action_move_create()
+
+    @api.model
+    def _prepare_refund(self, invoice, date_invoice=None,
+                        date=None, description=None, journal_id=None):
+        vals = super(AccountInvoice, self)._prepare_refund(
+            invoice, date_invoice, date, description, journal_id)
+
+        if invoice.type in ['in_invoice', 'in_refund']:
+            vals['check_total'] = invoice.check_total
```
Comment (nikul-serpentcs) [inline account_invoice_check_total/models/account_invoice.py:42]: @rven IMO here `invoice.check_total or 0.0`

**Review by nikul-serpentcs — COMMENTED** [review]
Improve Code

File: account_invoice_check_total/models/account_invoice.py (hunk 4 above)
Comment (rven) [reply account_invoice_check_total/models/account_invoice.py:42]: @nikul-serpentcs 
Store float value. No NULL value support. If value is not set it returns 0.0
see https://odoo-new-api-guide-line.readthedocs.io/en/latest/fields.html

File: account_invoice_check_total/__manifest__.py (hunk 2 above)
Comment (rven) [reply account_invoice_check_total/__manifest__.py]: @nikul-serpentcs Done

File: account_invoice_check_total/__manifest__.py (hunk 3 above)
Comment (rven) [reply account_invoice_check_total/__manifest__.py:15]: @nikul-serpentcs Done

**Comment by tafaRU** [comment]
@pedrobaeza, what about my https://github.com/OCA/account-invoicing/pull/430#issuecomment-426269737? Thanks.

**Comment by tafaRU** [comment]
@rven, please restart Travis build jobs now that branch `12.0` has been created.

**Pull request merged**